front/components/data_source/RequestDataSourcesModal.tsx

```diff
@@ -0,0 +1,146 @@
+import { Button, DropdownMenu, Modal, TextArea } from "@dust-tt/sparkle";
+import React, { useState } from "react";
+
+import { CONNECTOR_CONFIGURATIONS } from "@app/lib/connector_providers";
+import { sendEmail } from "@app/lib/email";
+import type { DataSourceIntegration } from "@app/pages/w/[wId]/builder/data-sources/managed";
+
+type RequestDataSourceProps = {
+  isOpen: boolean;
+  onClose: () => void;
+  dataSourceIntegrations: DataSourceIntegration[];
+  currentUserEmail: string;
+};
+
+async function sendRequestDataSourceEmail(
+  email: string,
+  emailContent: string,
+  ccEmail?: string
+) {
+  const mail = {
+    from: {
+      name: "Dust team",
+      email: "team@dust.tt",
+    },
+    subject: `[Dust] Request Data source`,
+    text: emailContent,
+  };
+  if (ccEmail) {
+    await sendEmail(email, mail, [ccEmail]);
+  } else {
+    await sendEmail(email, mail);
+  }
+}
+
+export function RequestDataSourcesModal({
+  isOpen,
+  onClose,
+  dataSourceIntegrations,
+  currentUserEmail,
+}: RequestDataSourceProps) {
+  const [selectedDataSourceIntegration, setSelectedDataSourceIntegration] =
+    useState<DataSourceIntegration | null>(null);
+  const [message, setMessage] = useState("");
```

Comment:
Maybe reset the message and selectedDataSourceIntegration when closing the modal ?